### PR TITLE
Support DeepSpeed <0.7.0

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/issues/13191

### Does your PR introduce any breaking changes? If yes, please list them.

Not by us, but DeepSpeed>=0.6.5 is not supported in Lite.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @awaelchli @rohitgr7 @akihironitta @borda

### CODE REVIEW

File: tests/tests_pytorch/lite/test_lite.py [hunk 1]
```diff
@@ -400,6 +401,8 @@ def test_autocast():
     lite._precision_plugin.forward_context().__exit__.assert_called()
 
 
+# https://github.com/microsoft/DeepSpeed/issues/2139
+@pytest.mark.skipif(_RequirementAvailable("deepspeed>=0.6.5"), reason="Lite does not support 0.6.5")
```
Comment (carmocca) [inline tests/tests_pytorch/lite/test_lite.py:405]: @awaelchli We should also have a version check in the Lite code so that users don't think we have bugs.

Comment (awaelchli) [reply tests/tests_pytorch/lite/test_lite.py]: Added that. Note that the test changes I made require your other PR that fixes the Lite test for <0.6.5

**Review by awaelchli — APPROVED** [review]

File: .azure/gpu-tests.yml [hunk 2]
```diff
@@ -73,7 +73,7 @@ jobs:
         CUDA_VERSION_MM=$(python -c "import torch ; print(''.join(map(str, torch.version.cuda.split('.')[:2])))")
         pip install "bagua-cuda$CUDA_VERSION_MM>=0.9.0"
         pip install -e .[strategies]
-        pip install deepspeed==0.6.4  # TODO: remove when docker images are upgraded
+        pip install deepspeed>0.6.4  # TODO: remove when docker images are upgraded
```
Comment (awaelchli) [inline .azure/gpu-tests.yml:76]: @carmocca the GPU CI passed. Should we address this TODO now or leave it like this?

Comment (carmocca) [reply .azure/gpu-tests.yml:76]: I would leave it like this. When the PR is merged and the dockers get rebuilt overnight, we can remove it without fear of things breaking

Comment (carmocca) [reply .azure/gpu-tests.yml:76]: Alternatively, if we remove it now, even though we validated that this worked inside the PR, there's a window (from the time of merge to the docker rebuilds) where things could go bananas :banana:

Comment (awaelchli) [reply .azure/gpu-tests.yml:76]: Ok. I was just wondering if there was an action to take before merge. 
I like bananas though.

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — APPROVED** [review]
